code/modules/fulton/fulton.dm

```diff
@@ -0,0 +1,163 @@
+var/list/normal_extraction_beacons = list()
+var/list/total_extraction_beacons = list()
+var/list/extraction_appends = list("-AAAAAAAAAAAAAAAAAUGH", "-AAAAAAAAAAAHHHHHHHHHH")
+
+/obj/item/weapon/extraction_pack
+	name = "fulton recovery pack"
+	desc = "A balloon that can be used to extract a target to a Fulton Recovery Beacon. Anything not bolted down can be moved. Link the pack to a beacon by using the pack in hand."
+	icon = 'icons/obj/fulton.dmi'
+	icon_state = "extraction_pack"
+	var/obj/machinery/extraction_point/beacon
+	var/is_syndie = 0
+	var/uses_left = 12
+
+/obj/item/weapon/extraction_pack/examine(mob/user)
+	. = ..()
+	user.show_message("It has [uses_left] uses remaining.", 1)
+
+/obj/item/weapon/extraction_pack/advanced
+	name = "advanced fulton recovery pack"
+	uses_left = 24
+
+/obj/item/weapon/extraction_pack/super
+	name = "super fulton recovery pack"
+	uses_left = 48
+
+/obj/item/weapon/extraction_pack/syndicate
+	name = "syndicate fulton recovery pack"
+	is_syndie = 1
+	uses_left = 9999
+
+/obj/item/weapon/extraction_pack/attack_self(mob/user)
+	if(!total_extraction_beacons.len)
+		user << "There are no extraction beacons in existence!"
+		return
+	else
+		var/A
+		if(is_syndie)
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in total_extraction_beacons
+		else
+			A = input("Select a beacon to connect to", "Fulton Recovery Pack", A) in normal_extraction_beacons
+		if(!A)
+			return
+		beacon = A
+
+/obj/item/weapon/extraction_pack/afterattack(atom/movable/A, mob/living/carbon/human/user, flag, params)
+	if(!beacon)
+		user << "[src] is not linked to a beacon, and cannot be used."
+		return
+	if(!flag)
+		return
+	if(!istype(A))
+		return
+	else
+		if(A.loc == user || A == user) // no extracting stuff you're holding in your hands/yourself
+			return
+		if(A.anchored)
+			return
+		user << "<span class='notice'>You start attaching the pack to [A]...</span>"
+		if(do_after(user,50,target=A))
+			if(A.anchored)
+				return
+			user << "<span class='notice'>You attach the pack to [A] and activate it.</span>"
+			uses_left--
+			if(uses_left <= 0)
+				user.drop_item(src)
+				loc = A
+			var/image/balloon
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.Weaken(16) // Keep them from moving during the duration of the extraction
+				M.buckled = 0 // Unbuckle them to prevent anchoring problems
+			else
+				A.anchored = 1
+				A.density = 0
+			var/obj/effect/extraction_holder/holder_obj = new(A.loc)
+			holder_obj.appearance = A.appearance
+			A.loc = holder_obj
+			balloon = image(icon,"extraction_balloon")
+			balloon.pixel_y = 10
+			balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
```

Comment:
`/image` is a child of `/datum`. Make this balloon stuff a subtype of `/image` and then give extraction packs another `var/global` pointing to the balloon image.